src/windows/audio.c

```diff
@@ -401,23 +428,37 @@ static bool audio_handle_device_change(MTY_Audio *ctx)
 
 uint32_t MTY_AudioGetQueued(MTY_Audio *ctx)
 {
-	return lrint((float) audio_get_queued_frames(ctx) / ((float) ctx->sample_rate / 1000.0f));
+	uint32_t queued = 0;
+	audio_get_queued_frames(ctx, &queued);
+	return lrint((float) queued / ((float) ctx->sample_rate / 1000.0f));
 }
 
 void MTY_AudioQueue(MTY_Audio *ctx, const int16_t *frames, uint32_t count)
 {
 	if (!audio_handle_device_change(ctx))
 		return;
 
-	uint32_t queued = audio_get_queued_frames(ctx);
+	uint32_t queued = 0;
+	HRESULT e = audio_get_queued_frames(ctx, &queued);
+	if (e != S_OK) {
+		MTY_Log("\"audio_get_queued_frames\" returned 0x%X. Re-initializing audio device", e);
+
+		e = audio_reinit_device(ctx);
+		if (e != S_OK) {
+			MTY_Log("\"audio_reinit_device\" returned 0x%X, failed to re-initialize audio device", e);
+			return;
+		}
+
+		e = audio_get_queued_frames(ctx, &queued);
```

Comment:
This `e` doesn't get used.

```suggestion
		audio_get_queued_frames(ctx, &queued);
```